src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.Unix.cs

```diff
@@ -143,6 +149,56 @@ private static Version GetOSXKernelVersion()
             return new Version(0, 0, 0);
         }
 
+        private static Version GetOSXProductVersion()
+        {
+            try
+            {
+                if (IsOSX)
+                {
+                    // <plist version="1.0">
+                    // <dict>
+                    //         <key>ProductBuildVersion</key>
+                    //         <string>17A330h</string>
+                    //         <key>ProductCopyright</key>
+                    //         <string>1983-2017 Apple Inc.</string>
+                    //         <key>ProductName</key>
+                    //         <string>Mac OS X</string>
+                    //         <key>ProductUserVisibleVersion</key>
+                    //         <string>10.13</string>
+                    //         <key>ProductVersion</key>
+                    //         <string>10.13</string>
+                    // </dict>
+                    // </plist>
+
+                    XElement dict = XDocument.Load("/System/Library/CoreServices/SystemVersion.plist").Root.Element("dict");
+                    if (dict != null)
+                    {
+                        foreach (XElement key in dict.Elements("key"))
+                        {
+                            if ("ProductVersion".Equals(key.Value))
+                            {
+                                XElement stringElement = key.NextNode as XElement;
+                                if (stringElement != null && stringElement.Name.LocalName.Equals("string"))
+                                {
+                                    string versionString = stringElement.Value;
+                                    if (versionString != null)
+                                    {
+                                        return Version.Parse(versionString);
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            catch
+            {
+            }
+
+            // In case of exception or couldn't get the version 
+            return new Version(0, 0, 0);
```

Comment:
I am not expecting to fail. and if we run into one of such cases I can implement another fallback for that. but so far I am not seeing we need this fallback path.